lib/diagnostic.js

```diff
@@ -38,10 +45,21 @@ const select = function select(array, callback) {
 // Question 5
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
-const first = function first(array, callback) {
 
+
+const first = function (array, callback) {
+    if (typeof array[i] !=== 'undefined') {
+        return undefined;
+    }
+
+    for (let i = 0; i < arr.length; i++) {
+        if (i in this. && callback (previousValue,this[i],a)) return this[i];
```

Comment:
The `i in this` bit is unnecessary (and I'm not sure it's valid syntax). The callback also needs only to operate on any given element.
i.e.
```
for (let i = 0; i < arr.length; i++) {
  if (callback(array[i])) {
    return array[i];
  }
}
```